class-es-wp-query-wrapper.php

```diff
@@ -73,14 +74,86 @@ protected function set_posts( $q, $es_response ) {
 							$post__in = implode( ',', $post_ids );
 							$this->posts = $wpdb->get_results( "SELECT $wpdb->posts.* FROM $wpdb->posts WHERE ID IN ($post__in) ORDER BY FIELD( {$wpdb->posts}.ID, $post__in )" );
 						}
-						return;
 					}
+
+					$this->posts = $this->post_query_sort_handler( $this->posts, $q );
 			}
 		} else {
 			$this->posts = array();
 		}
 	}
 
+	/**
+	 * Post query sort handler
+	 * Handle sorting by `post__in`, `post__in` and `post_parent__in`.
+	 *
+	 * @param  array $posts Query result posts.
+	 * @param  array $query Initial query.
+	 * @return array Sorted posts.
+	 */
+	protected function post_query_sort_handler( $posts, $query ) {
+		if ( empty( $query['orderby'] ) ) {
+			return $posts;
+		}
+
+		// Determine the key to sort by.
+		switch ( $query['orderby'] ) {
+			case 'post__in' :
+				$key = 'ID';
+				break;
+
+			case 'post_name__in' :
+				$key = 'post_name';
+				break;
+
+			case 'post_parent__in' :
+				$key = 'post_parent';
+				break;
+
+			default :
+				return $posts;
+		}
+
+		// Flip the order to allow retrieval by index.
+		$order = array_flip( $query[ $query['orderby'] ] );
+		$use_source = apply_filters( 'es_query_use_source', false );
+
+		usort( $posts, function( $a, $b ) use ( $order, $key, $use_source ) {
+			// Add support for using the Elasticsearch _source field.
+			if ( $use_source ) {
+				// Elasticsearch stores the `ID` field as `post_id`.
```

Comment:
Not necessarily; VIP's ES mapping != SearchPress'. All fields need to be "translated" to the adapter's map, e.g. `$this->es_map( 'post_id' )` to get the right field. Same goes for the switch above where `$key` gets set.